doc/rm/register_tool/index.md

```diff
@@ -571,6 +571,148 @@ There is no way to guarantee that hardware is doing the right thing, but at leas
 
 Similarly it is legal to set `hwqe` true for any of these register types if the clearing wants to be monitored by the hardware outside.
 
+## Shadow Registers
+
+If in the `.hjson` file the optional key `shadowed` is set to `true` for a particular register group, the register tool will implement the corresponding registers including all fields as shadow registers using the `prim_subreg_shadow` module.
+The following section describes the concept in detail.
+
+### Overview
+
+For Comportable designs used in sensitive security environments, for example AES or HMAC peripherals, there may be a need to guard against fault injection attacks.
+One of the surfaces under threat are critical configuration settings that may be used to control keys, region access control and system security configuration.
+These registers do not typically contain secret values, but can be vulnerable to fault attacks if a fault can alter the state of a register to a specific state.
+
+The fault attacks targeting these registers can come in a few forms:
+- A fault attack directly on the storage element.
+- A fault attack on peripheral logic (for example resets) that cause the registers to revert to a default state.
+- A fault attack during the process of the write to disrupt the actual written value.
+
+Shadow registers are a mechanism to guard sensitive registers against this specific type of attack.
+
+### Description
+
+The diagram below visualizes the internals of the `prim_subreg_shadow` module used to implement shadow registers.
+
+![subreg_shadow](subreg_shadow.svg)
+
+When compared to a normal register, the `prim_subreg_shadow` module is comprised of two additional components:
+- Staged register, and
+- Shadow register.
+- The committed register is the same existing storage.
+
+Under the hood, these registers are implemented using the `prim_subreg` module.
+This means a shadow register can implement any of the types described in [this section]({{< relref "doc/rm/register_tool/index.md#register-definitions-per-type" >}}).
+
+The general behavior of updating a shadow register is as follows:
+1. Software performs a first write to the register.
+
+   The staged register takes on the new value.
+   The committed register is **not** updated, so the hardware still sees the old value.
+
+1. Software performs a second write to the register (same address):
+
+   - If the data of the second write matches the value in the staged register, **both** the shadow register and the committed register are updated.
+   - If the data of the second write does **not** match the contents stored in the staged register, an update error is signaled to the hardware with a pulse on `err_update`.
+
+   The phase is tracked internally to the module.
+   If needed, software can put the shadow register back into the first phase by performing a read operation.
+   If the register is of type **RO**, software cannot interfere with the update process and read operations do not clear the phase tracker.
```

Comment:
Would prefer `RO` over **RO**.

Does a shadowed `RO` even make sense?

Reply:
I agree on `RO` vs **RO**. This is changed.

Reply:
> Does a shadowed `RO` even make sense?

I can't really imagine a case where it would be useful. I could add a check in the regtool to prevent this. But what I really care about is supporting the same types as for the normal registers, mainly to avoid documenting the supported types again in the shadow register section.